modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/view_object_entries.jsp

```diff
@@ -18,17 +18,30 @@
 
 <%
 ViewObjectEntriesDisplayContext viewObjectEntriesDisplayContext = (ViewObjectEntriesDisplayContext)request.getAttribute(WebKeys.PORTLET_DISPLAY_CONTEXT);
+
+ObjectDefinition objectDefinition = viewObjectEntriesDisplayContext.getObjectDefinition();
 %>
 
-<clay:headless-data-set-display
-	apiURL="<%= viewObjectEntriesDisplayContext.getAPIURL() %>"
-	clayDataSetActionDropdownItems="<%= viewObjectEntriesDisplayContext.getClayDataSetActionDropdownItems() %>"
-	creationMenu="<%= viewObjectEntriesDisplayContext.getCreationMenu() %>"
-	formName="fm"
-	id="<%= viewObjectEntriesDisplayContext.getClayHeadlessDataSetDisplayId() %>"
-	itemsPerPage="<%= 20 %>"
-	namespace="<%= liferayPortletResponse.getNamespace() %>"
-	pageNumber="<%= 1 %>"
-	portletURL="<%= liferayPortletResponse.createRenderURL() %>"
-	style="fluid"
-/>
\ No newline at end of file
+<c:choose>
+	<c:when test="<%= objectDefinition.isPortlet() %>">
+		<clay:headless-data-set-display
+			apiURL="<%= viewObjectEntriesDisplayContext.getAPIURL() %>"
+			clayDataSetActionDropdownItems="<%= viewObjectEntriesDisplayContext.getClayDataSetActionDropdownItems() %>"
+			creationMenu="<%= viewObjectEntriesDisplayContext.getCreationMenu() %>"
+			formName="fm"
+			id="<%= viewObjectEntriesDisplayContext.getClayHeadlessDataSetDisplayId() %>"
+			itemsPerPage="<%= 20 %>"
+			namespace="<%= liferayPortletResponse.getNamespace() %>"
+			pageNumber="<%= 1 %>"
+			portletURL="<%= liferayPortletResponse.createRenderURL() %>"
+			style="fluid"
+		/>
+	</c:when>
+	<c:otherwise>
+		<clay:alert
+			displayType="warning"
+			message="This is Object is not available or it was not published."
```

Comment:
We can define this message in the `portal-language-lang` and use it like this example:

[view_flat_organizations.jsp:L144](https://github.com/liferay/liferay-portal/blob/master/modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_organizations.jsp#L144-L146)

This way we can make this message localizable

Reply:
Also, can you rephrase for something like "This object is not available"

Reply:
Ok @gabrielwas, I'll do it thanks! 😃